### ISSUE TITLE
Action models always giving same output

### ISSUE THREAD

**Issue by sanketsans** [epic-kitchens/C1-Action-Recognition-TSN-TRN-TSM#16]
I am trying to test the models on a personal egocentric dataset. Instead of creating a video dataset, I extract frames from the videos and stack them together (tested with frame_count 8 and 25) and feed it to model (TSN and TSM). 
This is my code below : 
`folder = '/data/sample/'
    transforms = transforms.Compose([transforms.CenterCrop(224),
                                    transforms.ToTensor()])

    dict = torch.load('/data/tsn_rgb.ckpt', map_location="cpu")
    cfg =  OmegaConf.create(dict["hyper_parameters"])
    OmegaConf.set_struct(cfg, False)

    cfg.data._root_gulp_dir = os.getcwd()  # set default root gulp dir to prevent
    # exceptions on instantiating the EpicActionRecognitionSystem
    data_dir_key = f"test_gulp_dir"
    cfg.data[data_dir_key] = folder
    cfg.trainer.accelerator = None

    system = EpicActionRecognitionSystem(cfg)
    system.load_state_dict(dict["state_dict"])

    Img = None
    for i in range(1, 26):
        img = transforms(Image.open(folder + 'img_' + str(i) + '.jpg')).unsqueeze(dim=0)
        if i > 1 :
            Img = torch.cat((Img, img), dim=0)
        else:
            Img = img
    #
    Img = Img.unsqueeze(dim=0)
    print(Img.shape) # torch.Size([1, 25, 3, 224, 224])
    print(system)
    out = system(Img)
    print(out.shape)
    v, n = out[:, :97], out[:, 97:]
    print(v.shape, n.shape)
    print(torch.mean(v), torch.mean(n))`

This is my config : 
`{'modality': 'RGB', 'seed': 42, 'data': {'frame_count': 8, 'test_frame_count': 25, 'segment_length': 1, 'train_gulp_dir': '${data._root_gulp_dir}/rgb_train', 'val_gulp_dir': '${data._root_gulp_dir}/rgb_validation', 'test_gulp_dir': '/data/sample/', 'worker_count': 40, 'pin_memory': True, 'preprocessing': {'bgr': False, 'rescale': True, 'input_size': 224, 'scale_size': 256, 'mean': [0.485, 0.456, 0.406], 'std': [0.485, 0.456, 0.406]}, 'train_augmentation': {'multiscale_crop_scales': [1, 0.875, 0.75, 0.66]}, 'test_augmentation': {'rescale_size': 256}, '_root_gulp_dir': '/home/sanketthakur/Documents/gaze_pred/C1-Action-Recognition-TSN-TRN-TSM'}, 'model': {'type': 'TSN', 'backbone': 'resnet50', 'pretrained': 'imagenet', 'dropout': 0.7, 'partial_bn': True}, 'learning': {'batch_size': 4, 'optimizer': {'type': 'SGD', 'momentum': 0.9, 'weight_decay': 0.0005}, 'lr': 0.01, 'lr_scheduler': {'type': 'StepLR', 'gamma': 0.1, 'epochs': [20, 40]}}, 'trainer': {'gradient_clip_val': 20, 'max_epochs': 80, 'weights_summary': 'full', 'benchmark': True, 'terminate_on_nan': True, 'distributed_backend': 'dp', 'gpus': 0, 'accumulate_grad_batches': 2, 'accelerator': None}}`

The network always predicts verb_id as 0 and noun_id as 1. I am not sure, if I am doing something wrong here. Any help is appreciated. 
Thanks.

**Comment by willprice** [comment]
Do you normalise your input to the network?

**Comment by sanketsans** [comment]
yes, I normalize them with the mean and std given : `mean = [0.485, 0.456, 0.406] and std = [0.485, 0.456, 0.406]` 

Overall my transform is : 
```
transforms = transforms.Compose([transforms.CenterCrop(224),
                                    transforms.ToTensor(),
                                    transforms.Normalize(list(cfg.data.preprocessing.mean), list(cfg.data.preprocessing.std))])
```

I make sure to split the last tensor(397) into verbs (0:96) and noun (97:), and then I take the argmax of each. But unfortunately I always get 0 for verb and 1 for noun as outputs

**Comment by sanketsans** [comment]
I also tried changing to the transforms provided in the codebase. i am now following the same methods. 
`channel_count = (
        3 if cfg.modality == "RGB" else 2 * cfg.data.segment_length
    )
    common_transform = Compose(
        [
            Stack(
                bgr=cfg.modality == "RGB"
                and cfg.data.preprocessing.get("bgr", False)
            ),
            ToTorchFormatTensor(div=cfg.data.preprocessing.rescale),
            GroupNormalize(
                mean=list(cfg.data.preprocessing.mean),
                std=list(cfg.data.preprocessing.std),
            ),
            ExtractTimeFromChannel(channel_count),
        ]
    )
    test_transform = Compose(
        [
            GroupScale(cfg.data.test_augmentation.rescale_size),
            GroupCenterCrop(cfg.data.preprocessing.input_size),
            common_transform,
        ]
    )

    Img = [Image.open(folder + 'img_1.jpg')]

    for i in range(2, 25):
        Img.append(Image.open(folder + 'img_' + str(i) + '.jpg'))

    Img = test_transform(Img)
    Img = Img.unsqueeze(dim=0)
    print(Img.shape)
    outputs: Dict[str, Tensor] = system.forward_tasks(Img)
    tasks = {
        task: {
            # "output": outputs[task],
            "preds": outputs[task].argmax(-1),
            # "weight": 1,
        }
        for task in ["verb", "noun"]
    }
    print(tasks)`

I have tried TSN and TSM model after loading their weights from the checkpoints provided  here [C1-Action-Recognition-TSN-TRN-TSM ](https://github.com/epic-kitchens/C1-Action-Recognition-TSN-TRN-TSM#pretrained-models ) . Still I always get 0 for my verb index. I am trying it on similar set of actions as in epic-kitchens but different videos.

**Comment by willprice** [comment]
Have you checked that you can reproduce the results on the EPIC-KITCHENS dataset? Might be best to try that first to validate that you have the preprocessing etc set up correctly first before trying to apply them to your own videos.

**Comment by sanketsans** [comment]
Yes, It seems to work now but better with 24 non-overlapping frames. I am using the same transform.
